Fix #17805 Submit button with formaction outside of form tag

diego-mathis opened the pull request:
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | 
| Fixed issues  | #17805

diego-mathis committed:
Fix #17805 Submit button with formaction outside of form tag

diego-mathis committed:
Add test for submit button outside of form tag using form attribute on button.
Fix form attribute outside of form tag not used.

My6UoT9 commented:
This does not work correctly, at least the gii generator is broken with this patch.
Still fixing it, but no eta.

diego-mathis committed:
Fix #17805 another try, tests to be done.
form is actually more difficult to implement it seems, it is also related to yii.js, but it should at least work when data-form is used now.

diego-mathis committed:
Merge branch 'yiisoft:master' into master
